api/client/swarm/init.go

```diff
@@ -45,19 +54,61 @@ func newInitCommand(dockerCli *client.DockerCli) *cobra.Command {
 func runInit(dockerCli *client.DockerCli, flags *pflag.FlagSet, opts initOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
+	spec := opts.swarmOptions.ToSpec()
+
+	var secret string
+	// If no secret was specified, we create a random one
+	if !flags.Changed("secret") {
+		var secretBytes [generatedSecretEntropyBytes]byte
+
+		if _, err := cryptorand.Read(secretBytes[:]); err != nil {
+			panic(fmt.Errorf("failed to read random bytes: %v", err))
+		}
+
+		var nn big.Int
+		nn.SetBytes(secretBytes[:])
+		secret = fmt.Sprintf("%0[1]*s", maxGeneratedSecretLength, nn.Text(generatedSecretBase))
+
+		fmt.Fprintf(dockerCli.Out(), "No node joining secret provided. Generated random secret:\n\t%s\n\n", secret)
```

Comment:
> In the secret flag usage text we say "Set secret value needed to accept nodes into cluster". Is that wrong? Should it be "secret value needed to join a cluster" ?

Yeah, the text should be changed to this. Acceptance is a separate step (as I just learned yesterday).

I'll change this to `No --secret provided for now`, but just FYI, based on discussions with @aluzzardi and @diogomonica, it sounds like we will be removing the ability to set a user-specified secret in a followup PR, and going with generated secrets only. This will allow us to make the secret more discoverable because we won't have to worry about exposing passwords from the API. Here is a mockup: https://gist.github.com/aaronlehmann/a919b014cf43fc46ab046a84d06bd097